code/modules/mob/living/carbon/xenomorph/life.dm

```diff
@@ -33,6 +33,10 @@
 	update_action_button_icons()
 	update_icons()
 
+	if (!away_ssd_checked && world.time - away_time > XENO_AFK_TIMER)
```

Comment:
instead of checking this every life tick you should use a timer triggered by client/Del()

Reply:
then you dont need the away_ssd_checked var